app/stac_api/serializers/item.py

```diff
@@ -241,6 +242,37 @@ def validate_type(self, value):
             )
         return normalize_and_validate_media_type(value)
 
+    def _validate_href_field(self, attrs):
+        """Only allow the href field if the collection allows for external assets
+
+        Raise an exception, this replicates the previous behaviour when href
+        was always read_only
+        """
+        # the href field is translated to the file field here
+        if 'file' in attrs:
+            if self.collection:
+                collection = self.collection
+            else:
+                raise LookupError("No collection defined.")
+
+            if not collection.allow_external_assets:
+                logger.info(
+                    'Attempted external asset upload with no permission',
+                    extra={
+                        'collection': self.collection, 'attrs': attrs
+                    }
+                )
+                errors = {'href': _("Found read-only property in payload")}
+                raise serializers.ValidationError(code="payload", detail=errors)
+
+            try:
+                validate_href_url(attrs['file'], collection)
+                if self.context.get("validate_href_reachability", True):
```

Comment:
Maybe just add a little comment here why we would omit the check, so that a future reader can connect this to the bulk upload